cirq-core/cirq/transformers/transformer_api.py

```diff
@@ -218,77 +218,41 @@ class TransformerContext:
     ignore_tags: Tuple[Hashable, ...] = ()
 
 
-TRANSFORMER = Callable[['cirq.AbstractCircuit', TransformerContext], 'cirq.AbstractCircuit']
-_TRANSFORMER_TYPE = Callable[['cirq.AbstractCircuit', TransformerContext], CIRCUIT_TYPE]
-
-
-def _transform_and_log(
-    func: _TRANSFORMER_TYPE[CIRCUIT_TYPE],
-    transformer_name: str,
-    circuit: 'cirq.AbstractCircuit',
-    context: TransformerContext,
-) -> CIRCUIT_TYPE:
-    """Helper to log initial and final circuits before and after calling the transformer."""
-
-    context.logger.register_initial(circuit, transformer_name)
-    transformed_circuit = func(circuit, context)
-    context.logger.register_final(transformed_circuit, transformer_name)
-    return transformed_circuit
-
+class TRANSFORMER(Protocol):
+    def __call__(
+        self, circuit: 'cirq.AbstractCircuit', context: TransformerContext
+    ) -> 'cirq.AbstractCircuit':
+        ...
 
-def _transformer_class(
-    cls: Type[_TRANSFORMER_TYPE[CIRCUIT_TYPE]],
-) -> Type[_TRANSFORMER_TYPE[CIRCUIT_TYPE]]:
-    old_func = cls.__call__
 
-    def transformer_with_logging_cls(
-        self: Type[_TRANSFORMER_TYPE[CIRCUIT_TYPE]],
-        circuit: 'cirq.AbstractCircuit',
-        context: TransformerContext,
-    ) -> CIRCUIT_TYPE:
-        def call_old_func(c: 'cirq.AbstractCircuit', ct: TransformerContext) -> CIRCUIT_TYPE:
-            return old_func(self, c, ct)
-
-        return _transform_and_log(call_old_func, cls.__name__, circuit, context)
-
-    setattr(cls, '__call__', transformer_with_logging_cls)
-    return cls
-
-
-def _transformer_func(func: _TRANSFORMER_TYPE[CIRCUIT_TYPE]) -> _TRANSFORMER_TYPE[CIRCUIT_TYPE]:
-    @functools.wraps(func)
-    def transformer_with_logging_func(
-        circuit: 'cirq.AbstractCircuit',
-        context: TransformerContext,
-    ) -> CIRCUIT_TYPE:
-        return _transform_and_log(func, func.__name__, circuit, context)
-
-    return transformer_with_logging_func
+_TRANSFORMER_T = TypeVar('_TRANSFORMER_T', bound=TRANSFORMER)
+_TRANSFORMER_CLS_T = TypeVar('_TRANSFORMER_CLS_T', bound=Type[TRANSFORMER])
 
 
 @overload
-def transformer(cls_or_func: _TRANSFORMER_TYPE[CIRCUIT_TYPE]) -> _TRANSFORMER_TYPE[CIRCUIT_TYPE]:
+def transformer(cls_or_func: _TRANSFORMER_T) -> _TRANSFORMER_T:
     pass
 
 
 @overload
-def transformer(
-    cls_or_func: Type[_TRANSFORMER_TYPE[CIRCUIT_TYPE]],
-) -> Type[_TRANSFORMER_TYPE[CIRCUIT_TYPE]]:
+def transformer(cls_or_func: _TRANSFORMER_CLS_T) -> _TRANSFORMER_CLS_T:
     pass
 
 
 def transformer(cls_or_func: Any) -> Any:
     """Decorator to verify API and append logging functionality to transformer functions & classes.
 
-    The decorated function or class must satisfy
-    `Callable[[cirq.Circuit, cirq.TransformerContext], cirq.Circuit]` API. For Example:
+    A transformer is a callable that takes as inputs a cirq.AbstractCircuit and
```

Comment:
Add ticks around `cirq.<>` references in docs. 
```suggestion
    A transformer is a callable that takes as inputs a `cirq.AbstractCircuit` and
```

Reply:
Done.